app/src/main/java/org/fossasia/openevent/activities/SpeakersActivity.java

```diff
@@ -0,0 +1,98 @@
+package org.fossasia.openevent.activities;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.DefaultItemAnimator;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.Toolbar;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import org.fossasia.openevent.Adapters.SessionsAdapter;
+import org.fossasia.openevent.R;
+import org.fossasia.openevent.data.Session;
+import org.fossasia.openevent.data.Speaker;
+import org.fossasia.openevent.dbutils.DbSingleton;
+import org.fossasia.openevent.utils.SpeakerIntent;
+
+import java.text.ParseException;
+import java.util.List;
+
+/**
+ * Created by MananWason on 30-06-2015.
+ */
+public class SpeakersActivity extends AppCompatActivity {
+    SessionsAdapter sessionsAdapter;
+    private String speaker;
+    private Speaker clicked;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_speakers);
+        DbSingleton dbSingleton = DbSingleton.getInstance();
+        speaker = getIntent().getStringExtra("SPEAKER");
+        final Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar_speakers);
+        setSupportActionBar(toolbar);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+
+        try {
+            clicked = dbSingleton.getSpeakerbySpeakersname(speaker);
```

Comment:
I am searching by name here because I have the name of the speaker from the intent, so, I would have retrieve the id first from the db using the name, then only I can retrieve info using the id. So, to minimise code, I directly used name to get the speaker info.